### PR TITLE
[TestFix] Set1: Review of failing glusterd tests

### PR DESCRIPTION
**Pull request by leelavg**
Change-Id: Ia5a3f7df990360bbdc964c49496dea7399d499ab
Signed-off-by: Leela Venkaiah G <lgangava@redhat.com>

### CODE REVIEW

**Comment by leelavg** [comment]
Failing Tests:
1. tests/functional/glusterd/test_glusterd_start_stop_output.py: test_glusterd_start_stop_consistent_info
2. tests/functional/glusterd/test_remove_brick_after_restart_glusterd.py: test_remove_brick 
3. tests/functional/glusterd/test_probe_glusterd_down.py: test_peer_probe_when_glusterd_down
4. tests/functional/glusterd/test_gluster_does_not_do_posix_lock_when_multiple_client.py: test_gluster_does_not_do_posix_lock_when_multiple_client
---
1. test_glusterd_start_stop_consistent_info
- Remove `runs_on` as we aren't creating any volumes
- Reset glusterd reset count before and after the test
- 5/5 pass locally

2. test_remove_brick 
- Flaky test (2/5 pass locally), better to skip it

3. test_peer_probe_when_glusterd_down
- 5/5 pass locally
- Skip if failed again

4. test_gluster_does_not_do_posix_lock_when_multiple_client
- Incorrect test, this wouldn't pass even in local setup. Owner should review

General Notes:
- Most of the failing tests are due to glusterd restart and the first test mentioned here might be the root cause for that
- Will revisit remaining glusterd failing tests once after a full re-run with this PR

**Comment by leelavg** [comment]
pls review: @srijan-sivakumar

File: tests/functional/glusterd/test_glusterd_start_stop_output.py [hunk 1]
```diff
@@ -44,26 +56,25 @@ def test_glusterd_start_stop_consistent_info(self):
...
         # Stop glusterd to get the info for further comparison.
-        stop_output = self._run_command_in_servers(glusterd_stop_cmd)
+        stop_output = self._run_command_in_servers(self.glusterd_stop_cmd)
 
         # Start glusterd to get the info for further comparison.
-        start_output = self._run_command_in_servers(glusterd_start_cmd)
+        start_output = self._run_command_in_servers(self.glusterd_start_cmd)
 
         # Loop over the stop-stop cycle for 5 times and validate the output.
         for _ in range(5):
```
Comment (srijan-sivakumar) [inline tests/functional/glusterd/test_glusterd_start_stop_output.py:66]: I feel we should have a call to reset-failed before the for loop. From setup, we've already run start glusterd for 2 times. And If I remember it right, the burst limit is 6 within an hour.

**Review by srijan-sivakumar — CHANGES_REQUESTED** [review]

File: tests/functional/glusterd/test_glusterd_start_stop_output.py (hunk 1 above)
Comment (leelavg) [reply tests/functional/glusterd/test_glusterd_start_stop_output.py:66]: ack, i can just remove starting glusterd explicitly in setup and that'd be find as we'll not be breaching 6.

```
StartLimitBurst=6
StartLimitIntervalSec=3600
StartLimitInterval=3600
```

**Review by pranavprakash20 — APPROVED** [review]

**Review by srijan-sivakumar — APPROVED** [review]
LGTM

**Review by balakreddy — APPROVED** [review]
LGTM

**Pull request merged**